File: all/t75/c&cpp/answer.cpp
#include <iostream>
#include <filesystem>
#include <regex>
#include <vector>
#include <string>

namespace fs = std::filesystem;

void rename_files(const std::string& directory) {
    // Convert directory to Path object for easier handling
    fs::path dir_path(directory);

    if (!fs::exists(dir_path) || !fs::is_directory(dir_path)) {
        throw std::invalid_argument("The directory '" + directory + "' does not exist or is not a directory.");
    }

    // Get list of PNG files in the directory
    std::vector<fs::path> png_files;
    for (const auto& entry : fs::directory_iterator(dir_path)) {
        if (entry.is_regular_file() && entry.path().extension() == ".png") {
            png_files.push_back(entry.path());
        }
    }

    // Sort files alphabetically by their names
    std::sort(png_files.begin(), png_files.end());

    // Print the sorted list of files (for debugging)
    std::cout << "Sorted files:" << std::endl;
    for (const auto& file : png_files) {
        std::cout << file.filename().string() << std::endl;
    }

    // Rename files with sequence numbers
    std::string prev_base_name;
    int count = 1;

    std::regex re_pattern(R"((\d{3})(-\d)?(?=\.png$))");

    for (const auto& file : png_files) {
        // Extract base name without postfix and number
        std::string base_name = std::regex_replace(file.stem().string(), re_pattern, "");

        if (base_name != prev_base_name) {
            count = 1;
        }

        std::string new_filename = base_name + std::to_string(count).substr(1, 3) + ".png";
        fs::path new_file_path = file.parent_path() / new_filename;
        fs::rename(file, new_file_path);
        std::cout << "Renaming " << file.filename().string() << " to " << new_filename << std::endl;

        prev_base_name = base_name;
        ++count;
    }
}

File: all/t75/java/Answer.java
package org.real.temp;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.Comparator;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Answer {

    public static void main(String[] args) {
        renameFiles("path/to/directory");
    }

    /**
     * Renames PNG files in a specified directory by appending a sequence number to each file.
     * The files are sorted alphabetically, and each base name is assigned sequential numbers.
     *
     * @param directory The path to the directory containing PNG files to be renamed.
     */
    public static void renameFiles(String directory) {
        // Convert directory to Path object for easier handling
        Path dirPath = Paths.get(directory);

        if (!Files.isDirectory(dirPath)) {
            throw new IllegalArgumentException("The directory '" + directory + "' does not exist or is not a directory.");
        }

        // Get list of PNG files in the directory
        File[] pngFiles = dirPath.toFile().listFiles((dir, name) -> name.toLowerCase().endsWith(".png"));

        if (pngFiles == null) {
            throw new IllegalArgumentException("No PNG files found in the directory '" + directory + "'.");
        }

        // Sort files alphabetically by their names
        Arrays.sort(pngFiles, Comparator.comparing(File::getName));

        // Print the sorted list of files (for debugging)
        System.out.println("Sorted files:");
        for (File file : pngFiles) {
            System.out.println(file.getName());
        }

        // Rename files with sequence numbers
        String prevBaseName = null;
        int count = 1;

        Pattern pattern = Pattern.compile("(\\d{3})(-\\d)?(?=\\.png$)");

        for (File file : pngFiles) {
            // Extract base name without postfix and number
            String fileName = file.getName();
            String baseName = fileName.substring(0, fileName.lastIndexOf('.'));
            Matcher matcher = pattern.matcher(baseName);
            if (matcher.find()) {
                baseName = matcher.replaceAll("");
            }

            if (!baseName.equals(prevBaseName)) {
                count = 1;
            }

            String newFileName = baseName + String.format("%03d", count) + ".png";
            Path newFilePath = dirPath.resolve(newFileName);

            try {
                Files.move(file.toPath(), newFilePath);
                System.out.println("Renaming " + fileName + " to " + newFileName);
            } catch (IOException e) {
                System.err.println("Failed to rename " + fileName + " to " + newFileName);
                e.printStackTrace();
            }

            prevBaseName = baseName;
            count++;
        }
    }
}
